lib/pkg.ml

```diff
@@ -431,21 +440,27 @@ let infer_from_readme dir =
           | false -> None
           | true -> Some name))
 
-let infer_name dir =
+let try_infer_name dir =
   infer_from_dune_project dir >>= function
-  | Some n -> Ok n
+  | Some n -> Ok (Some n)
   | None -> (
       infer_from_opam_files dir >>= function
-      | Some n -> Ok n
+      | Some n -> Ok (Some n)
       | None -> (
           infer_from_readme dir >>= function
-          | Some n -> Ok n
-          | None ->
-              Logs.err (fun m ->
-                  m
-                    "cannot determine distribution name automatically: add \
-                     (name <name>) to dune-project");
-              exit 1))
+          | Some n -> Ok (Some n)
+          | None -> Ok None))
```

Comment:
What is the value of returning `Ok None` ? I think we can rewrite this function to return `(string, ['Msg of string]) result` so we can return the message directly instead of having `infer_name_err` on the side, and probably the èinfer_name` shouldn't return `exit 1` but let propagate the error value up up up?

Reply:
> What is the value of returning Ok None ?

The return type of this function gets propagated from the return type of the `infer_from...` functions. So the value of `Ok None` is the same as in the `infer_from...` functions:
there was no sys call error -> `Ok`;
there's not enough information in any of the resources to infer the name from them -> `None`

> and probably the `infer_name` shouldn't return `exit 1`

I was also wondering about that behaviour here. I've just preserved the behaviour. Since I didn't know why it was done that way in the first place, I didn't feel confident changing it. Do you happen to know why it was done like that?